txpool/tx_object_map.go

```diff
@@ -45,26 +51,53 @@ func (m *txObjectMap) Add(txObj *txObject, limitPerAccount int) error {
 		return nil
 	}
 
-	if m.quota[txObj.Origin()] >= limitPerAccount {
+	origin := originKey(txObj.Origin())
+	if m.quota[origin] >= limitPerAccount {
 		return errors.New("account quota exceeded")
 	}
 
-	if d := txObj.Delegator(); d != nil {
-		if m.quota[*d] >= limitPerAccount {
+	delegator := (*delegatorKey)(txObj.Delegator())
+	if delegator != nil {
+		if m.quota[*delegator] >= limitPerAccount {
 			return errors.New("delegator quota exceeded")
 		}
-		m.quota[*d]++
 	}
 
-	m.quota[txObj.Origin()]++
+	var (
+		cost  *big.Int
+		payer thor.Address
+	)
+
+	if txObj.Cost() != nil {
```

Comment:
Is it possible to flood the mempool with no-cost transactions ?

Reply:
Not possible at peer-to-peer level, possible at API level, won't propagate until executable.

Reply:
It seems like they wouldn't be added until executable right ? So we should be good.

However theres seems to be a different behaviour [here](https://github.com/vechain/thor/blob/cc2b3764839ce910d979cc6d1c283d60acfc17e6/txpool/tx_pool.go#L279-L294). Let's pair up and review it separately to this PR tomorrow perhaps ?